swupd/create_manifests.go

```diff
@@ -177,6 +192,70 @@ func processBundles(ui UpdateInfo, c config) ([]*Manifest, error) {
 	return newManifests, nil
 }
 
+func appendUniqueFiles(appendTo *Manifest, appendFrom *Manifest) {
+	for _, f := range appendFrom.Files {
+		if f.findFileNameInSlice(appendTo.Files) == nil {
+			if f.Name == indexBundle {
+				// this is generated new each time
+				continue
+			}
+			appendTo.Files = append(appendTo.Files, f)
+		}
+	}
+}
+
+func writeMansPopulateMoM(newManifests []*Manifest, newMoM *Manifest, out string) (*Manifest, error) {
+	var newFull *Manifest
+	var err error
+	// write manifests then add them to the MoM
+	for _, bMan := range newManifests {
+		// handle the full manifest last
+		if bMan.Name == "full" {
+			newFull = bMan
+			continue
+		}
+
+		// TODO: remove this after a format bump in Clear Linux
+		// this is a hack to set maximum contentsize to the incorrect maximum
+		// set in swupd-client v3.15.3
+		bMan.setMaxContentSizeHack()
+		// end hack
+
+		// sort by version then by filename, previously to this sort these bundles
+		// were sorted by file name only to make processing easier
+		bMan.sortFilesVersionName()
+		manPath := filepath.Join(out, "Manifest."+bMan.Name)
+		if err = bMan.WriteManifestFile(manPath); err != nil {
+			return nil, err
+		}
+
+		// add bundle to Manifest.MoM
+		if err = newMoM.createManifestRecord(out, manPath, newMoM.Header.Version); err != nil {
+			return nil, err
+		}
+	}
+
+	return newFull, nil
+}
+
+func aggregateManifests(newManifests []*Manifest, newMoM Manifest, version uint32, c config) ([]*Manifest, error) {
```

Comment:
I am not sure if this should take newMoM as a Manifest, a *Manifest, or just a slice of *File.